src/shared/flag-details/variance-select.tsx

```diff
@@ -1,24 +1,165 @@
-import * as React from "react";
-import Form, { FormItemProps } from "antd/lib/form";
+import React, { useState, useEffect } from "react";
+import { styled } from "onefx/lib/styletron-react";
+import Form, { FormProps, FormItemProps } from "antd/lib/form";
 import Select from "antd/lib/select";
+import InputNumber from "antd/lib/input-number";
+import Space from "antd/lib/space";
+import Row from "antd/lib/row";
+import Tag from "antd/lib/tag";
+import Col from "antd/lib/col";
+import { colors, VarIcon } from "@/shared/common/icons/var-icon";
 
 export const VarianceSelect: React.FC<{
   itemProps: FormItemProps;
+  form?: FormProps;
+  rolloutName?: string[];
   variance?: boolean[];
-  disabled?: boolean;
+  parentName?: string;
   loading?: boolean;
-}> = ({ itemProps, variance = [], disabled = true, loading }) => {
-  return (
+  isShowRollout?: boolean;
+}> = ({
+  form,
+  itemProps,
+  rolloutName,
+  variance = [],
+  parentName,
+  loading,
+  isShowRollout = false,
+}) => {
+  const [diffVariance, setDiffVariance] = useState({
+    total: 0,
+    currentVariances: [],
+  });
+
+  useEffect(() => {
+    if (form) {
+      const currentVariances = variance.map((_, k) =>
+        form.getFieldValue(
+          parentName ? [parentName, ...rolloutName, k] : [...rolloutName, k]
+        )
+      );
+      setDiffVariance({
+        total: currentVariances.reduce((a, b) => Number(a) + Number(b), 0),
+        currentVariances,
+      });
+    }
+  }, [variance]);
+
+  const renderVariancesSelect = () => (
     <Form.Item {...itemProps}>
-      <Select disabled={disabled} placeholder="Please select" loading={loading}>
+      <Select
+        placeholder="Please select"
+        loading={loading}
+        style={{ minWidth: "175px" }}
+      >
         {variance.map((v, i) => {
           return (
             <Select.Option key={String(v)} value={i}>
               {String(v)}
             </Select.Option>
           );
         })}
+        {isShowRollout && (
+          <Select.Option value={null}>a percentage rollout</Select.Option>
+        )}
       </Select>
     </Form.Item>
   );
+
+  return isShowRollout ? (
```

Comment:
avoid next conditions in the return. you could actually evaluate conditions in render function

```
if (condition) {
  return <>1</>
}

if (condition) {
  return <>2</>
}


return <>3</>

````